### PR TITLE
Rename cloud.zone to cloud.availability_zone

### PR DESCRIPTION
**Pull request by rakyll**
Only Google Cloud calls availability zones a zone. Renaming zone
to make the least common case an exception.

This change is a follow-up of https://github.com/open-telemetry/opentelemetry-specification/pull/1481.

### CODE REVIEW

File: semantic_conventions/resource/cloud.yaml [hunk 1]
```diff
@@ -33,15 +33,15 @@ groups:
...
         brief: >
           Cloud regions often have multiple, isolated locations known as zones
-          to increase availability. Zone represents the zone where the resource 
-          is running.
+          to increase availability. Availability zone represents the isolated
+          zone where the resource is running.
         note: >
-          AWS and Azure calls them availability zones or AZs.
-        examples: ['us-central1-a', 'us-east-1c']
+          Google Cloud calls availability zones zones.
```
Comment (reyang) [inline semantic_conventions/resource/cloud.yaml:43]: nit - avoid "zones zones"?

```suggestion
          Google Cloud calls them zones.
```

File: specification/resource/semantic_conventions/cloud.md [hunk 2]
```diff
@@ -12,10 +12,10 @@
...
 | `cloud.region` | string | The geographical region the resource is running. Refer to your provider's docs to see the available regions, for example [AWS regions](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/), [Azure regions](https://azure.microsoft.com/en-us/global-infrastructure/geographies/), or [Google Cloud regions](https://cloud.google.com/about/locations). | `us-central1`; `us-east-1` | No |
-| `cloud.zone` | string | Cloud regions often have multiple, isolated locations known as zones to increase availability. Zone represents the zone where the resource  is running. [1] | `us-central1-a`; `us-east-1c` | No |
+| `cloud.availability_zone` | string | Cloud regions often have multiple, isolated locations known as zones to increase availability. Availability zone represents the isolated zone where the resource is running. [1] | `us-east-1c` | No |
 | `cloud.infrastructure_service` | string | The cloud infrastructure resource in use. [2] | `aws_ec2`; `azure_vm`; `gcp_compute_engine` | No |
 
-**[1]:** AWS and Azure calls them availability zones or AZs.
+**[1]:** Google Cloud calls availability zones zones.
```
Comment (reyang) [inline specification/resource/semantic_conventions/cloud.md:18]: Or consider this:

```suggestion
**[1]:** Google Cloud calls availability zones **zones**.
```

**Review by reyang — APPROVED** [review]
It's like a political game, but your statement "only Google calls..." makes sense to me.
More importantly, I think availability zone is just a better term than "zone" since it is more specific.

**Review by iNikem — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

File: semantic_conventions/resource/cloud.yaml [hunk 3]
```diff
@@ -33,15 +33,15 @@ groups:
...
           [Google Cloud regions](https://cloud.google.com/about/locations).
         examples: ['us-central1', 'us-east-1']
-      - id: zone
+      - id: availability_zone
         type: string
         brief: >
           Cloud regions often have multiple, isolated locations known as zones
-          to increase availability. Zone represents the zone where the resource 
-          is running.
+          to increase availability. Availability zone represents the isolated
```
Comment (Oberon00) [inline semantic_conventions/resource/cloud.yaml:40]:
```suggestion
          to increase availability. Availability zone represents the
```

I would not repeat "isolated" here.

**Review by Oberon00 — APPROVED** [review]

**Review by alolita — APPROVED** [review]

File: semantic_conventions/resource/cloud.yaml (hunk 1 above)
Comment (arminru) [reply semantic_conventions/resource/cloud.yaml:43]: or
```suggestion
          Google Cloud calls availability zones just "zones".
```

**Review by arminru — APPROVED** [review]

File: specification/resource/semantic_conventions/cloud.md (hunk 2 above)
Comment (jsuereth) [reply specification/resource/semantic_conventions/cloud.md:18]: Or:  
`Availability zones are called "zones" on Google Cloud`

**Comment by rakyll** [comment]
PTAL

File: semantic_conventions/resource/cloud.yaml [hunk 4]
```diff
@@ -33,15 +33,15 @@ groups:
           [AWS regions](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/), [Azure regions](https://azure.microsoft.com/en-us/global-infrastructure/geographies/), or
           [Google Cloud regions](https://cloud.google.com/about/locations).
         examples: ['us-central1', 'us-east-1']
-      - id: zone
+      - id: availability_zone
```
Comment (yurishkuro) [inline semantic_conventions/resource/cloud.yaml:36]: Did we consider just `az`? This is a very long string that will be repeated in every span.

Comment (bogdandrutu) [reply semantic_conventions/resource/cloud.yaml:36]: It is in the resource not the Span :), but still worth considering `az` maybe

Comment (yurishkuro) [reply semantic_conventions/resource/cloud.yaml:36]: That distinction only matters at emission side, in the storage the resource attributes are usually denormalized into each span / log.

Comment (rakyll) [reply semantic_conventions/resource/cloud.yaml:36]: I think zone is more readable if we think availability_zone is too long. Happy to keep things as they are.

Comment (carlosalberto) [reply semantic_conventions/resource/cloud.yaml:36]: I'd suggest going with either `az` or `zone`.

Comment (Oberon00) [reply semantic_conventions/resource/cloud.yaml:36]: zone was the original name, before this PR. I suggested availability_zone, and I think it is more clear. az is not consistent with our other semantic convention names, which are all on the verboser side.

Comment (carlosalberto) [reply semantic_conventions/resource/cloud.yaml:36]: @Oberon00 Sounds like `az` is a no-go. Would `zone` be a good compromise? i.e. what we currently have ;)

Comment (Oberon00) [reply semantic_conventions/resource/cloud.yaml:36]: I would not say `az` is a no-go. Personally I think many of our conventions are more verbose than they should be.
I can also live with `zone`, i.e. what we currently have. I would have preferred `availability_zone` over it though.

Comment (bogdandrutu) [reply semantic_conventions/resource/cloud.yaml:36]: What I am hearing from this thread is that we do not have clear requirements for the semantic conventions names, and probably I would say that should be our next action item before merging or making a decision on this:
1. human readable vs storage space is one of the main topic to decide on.

Comment (carlosalberto) [reply semantic_conventions/resource/cloud.yaml:36]: Let's discuss human readable vs storage space in the next Spec meeting.

Comment (rakyll) [reply semantic_conventions/resource/cloud.yaml:36]: I'd prefer to keep it as zone if length is an issue. `az` is not readable enough.

**Comment by rakyll** [comment]
We discussed in the meeting and decided that readability matters more. We always have a way to compress the keys in the future on the wire and at rest if it's needed. This change should be good to be merged.

I will also send a change with recommendations on max key lengths as a guideline as we discussed earlier.

**Comment by rakyll** [comment]
@Oberon00 @iNikem @arminru @reyang Can we merge this change?

**Pull request merged**